modules/promotion/src/PromotionStorage.php

```diff
@@ -104,4 +104,61 @@ public function loadAvailable(OrderTypeInterface $order_type, StoreInterface $st
     return $promotions;
   }
 
+  /**
+   * Return active promotions that have passed their end date.
+   *
+   * @return \Drupal\commerce_promotion\Entity\PromotionInterface[]
+   *   The expired promotion entities.
+   */
+  public function loadExpired() {
+    $query = $this->getQuery();
+
+    $query
+      ->condition('end_date', gmdate('Y-m-d'), '<')
+      ->condition('status', TRUE);
+
+    $result = $query->execute();
+
+    if (empty($result)) {
+      return [];
+    }
+
+    return $this->loadMultiple($result);
+  }
+
+  /**
+   * Returns active promotions which have a met their maximum usage.
```

Comment:
Document blocks should only be on the interface, implementers just do `{@inheritdoc}`